src/Features/Core/Portable/GenerateMember/GenerateVariable/AbstractGenerateVariableService.CodeAction.cs

```diff
@@ -70,7 +70,7 @@ protected override Task<Document> GetChangedDocumentAsync(CancellationToken canc
                         accessibility: DetermineMaximalAccessibility(_state),
                         modifiers: new DeclarationModifiers(isStatic: _state.IsStatic, isUnsafe: generateUnsafe),
                         type: _state.TypeMemberType,
-                        returnsByRef: _returnsByRef,
+                        refKind: _returnsByRef ? RefKind.Ref : RefKind.None,
```

Comment:
is this sufficient?  is there a ref-readonly case this won't work for?

Reply:
Right now, we don't have the mechanism to guess that from syntax at call-site. I added #22385 to track that.

---
In reply to: [141180693](https://github.com/dotnet/roslyn/pull/22285#discussion_r141180693) [](ancestors = 141180693)